src/main/java/net/mcreator/element/parts/gui/Sprite.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2024, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.element.parts.gui;
+
+import net.mcreator.element.parts.procedure.Procedure;
+import net.mcreator.ui.workspace.resources.TextureType;
+import net.mcreator.ui.wysiwyg.WYSIWYGEditor;
+import net.mcreator.util.FilenameUtilsPatched;
+import net.mcreator.util.image.ImageUtils;
+import net.mcreator.workspace.Workspace;
+import net.mcreator.workspace.references.TextureReference;
+
+import javax.swing.*;
+import java.awt.*;
+
+public class Sprite extends GUIComponent {
+
+	@TextureReference(TextureType.SCREEN) public String sprite;
+
+	public int spritesCount;
+
+	public Procedure displayCondition;
+	public Procedure spriteIndex;

```

Comment:
I would recommend using NumberProcedure here, so either Procedure or fixed value can be used. This way one can re-use same sprite image for multiple component fixed icons that don't change.

Then also use NumberProcedureSelector and you will need some minor template adaptations.

Check existing use cases of NumberProcedure for inspiration